binary/test.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2020 Simon Biggs
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+import subprocess
+import sys
+import time
+import socket
+from contextlib import contextmanager
+
+import psutil
+
+
+HERE = pathlib.Path("__file__").parent.resolve()
+
+BUILD = HERE.joinpath("build")
+BUILD_DIST = BUILD.joinpath("dist")
+
+REPO_ROOT = HERE.parent
+CYPRESS = REPO_ROOT.joinpath("lib", "pymedphys", "tests", "e2e")
+
+
+def main():
+    if sys.platform == "win32":
```

Comment:
sys.platform is specified as a compiler define during the build configuration.
os.name checks whether certain os specific modules are available (e.g. posix, nt, ...)
platform.system() actually runs uname and potentially several other functions to determine the system type at run time.

is using sys.platform the most appropriate approach?
Is win32 a safe bet (will some systems come up with win64 or amd64?)?

Reply:
See https://docs.python.org/3/library/sys.html#sys.platform

Windows will always return `win32`. Arguably however `platform.system` is a nicer, easier to understand response. Usage of `sys.platform` is more knee-jerk habit at play.

A google search doesn't disagree with my usage here:

https://stackoverflow.com/a/11674977/3912576